charms/worker/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -776,9 +776,30 @@ def is_cluster_ready(self):
         return status.metadata and status.metadata.status.ready
 
     def check_k8sd_ready(self):
-        """Check if k8sd is ready."""
-        endpoint = "/cluster/1.0/ready"
-        self._send_request(endpoint, "GET", EmptyResponse)
+        """Check if k8sd is ready using various microcluster endpoints.
+
+        Raises:
+            K8sdConnectionError: If the response is Not Found on all endpoints.
+        """
+        ready_endpoints = ["/core/1.0/ready", "/cluster/1.0/ready"]
+        for endpoint in ready_endpoints:
+            try:
+                self._send_request(endpoint, "GET", EmptyResponse)
+                break
+            except InvalidResponseError as ex:
+                if ex.code == 404:
+                    logger.warning(
+                        "Encountered 404 while checking if micro-cluster is ready @ %s: %s",
+                        endpoint,
+                        ex,
+                    )
+                    # Try the next endpoint if the current one is not found
+                    continue
```

Comment:
k8sd responding with a 404 is its way of saying "I see that request -- but that doesn't exist".  A clear sign we should try another endpoint!

Reply:
So... this may be misleading the user into thinking something is wrong with the code.
Could the log be modified to mention this *could* be normal and we're cycling through possible endpoints?

Reply:
Maybe `Trying endpoint 1 of 2: ...`